crates/metaslang/cst/src/cursor.rs

```diff
@@ -21,11 +19,11 @@ struct PathAncestor {
 ///
 /// Nodes are visited in a DFS pre-order traversal.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Cursor {
+pub struct Cursor<T: KindTypes> {
     /// The parent path of this cursor
-    parent: Option<Rc<PathAncestor>>,
+    parent: Option<Rc<PathAncestor<T>>>,
```

Comment:
unrelated to this PR: is it intentional to wrap `PathAncestor` in another `Rc` given that it is already using `Rc`s internally?

Reply:
Yes, to maximise the sharing of the path to the root

Reply:
Makes sense. Thanks.